src/CMake/SetupCompilerOptions.cmake

```diff
@@ -31,3 +31,16 @@ endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 message(STATUS "Updated CMAKE_CXX_FLAGS to \"${CMAKE_CXX_FLAGS}\"")
+
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+	message(STATUS "Debug build: Updated CMAKE_C_FLAGS to \"${CMAKE_C_FLAGS}\"")
+	message(STATUS "Debug build: Updated CMAKE_CXX_FLAGS to \"${CMAKE_CXX_FLAGS}\"")
+endif()
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+	message(STATUS "Release build: Updated CMAKE_C_FLAGS to \"${CMAKE_C_FLAGS}\"")
+	message(STATUS "Release build: Updated CMAKE_CXX_FLAGS to \"${CMAKE_CXX_FLAGS}\"")
+endif()
```

Comment:
I'm wondering if we want users to experience a build failure if any warnings are encountered. I am thinking perhaps a better place for the `-Wall -Werror` flags is to add this to the variorum build that we use on Travis, so that warnings must be fixed (by developers) as part of their PR. Thoughts?